tests/test_lora.py

```diff
@@ -758,3 +758,13 @@ def test_lora_model_fsdp_init():
     model = fabric.setup(model)
     y = model(x)
     assert y.shape == torch.Size([2, 8, 512])
+
+    # verify that all the parameters, buffers and other attributes aren't on `meta` device
+    for m in model.modules():
+        for p_name, parameter in m.named_parameters():
+            assert not parameter.is_meta, f"Parameter `{p_name}` isn't materialized."
+        for b_name, buffer in m._buffers.items():
+            assert not buffer.is_meta, f"Buffer `{b_name}` isn't materialized."
+        for attr_name, attr_value in m.__dict__.items():
+            if isinstance(attr_value, torch.Tensor):
+                assert not attr_value.is_meta, f"Attribute `{attr_name}` isn't materialized."
```

Comment:
There's a simpler way to do this, which is to do

`model.to(device="cpu")  # or cuda`

If something is on meta, it will raise an error.

Reply:
Correct.
Plus with `model.to(device="cpu")` you will get something like:
> NotImplementedError: Cannot copy out of meta tensor; no data!

and good luck finding what exactly is wrong. Here you at least get the name of the non-materialized parameter/buffer/attribute.